sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/windowing/AfterPane.java

```diff
@@ -103,6 +103,11 @@ public void clear(TriggerContext c) throws Exception {
   }
 
   @Override
+  public boolean hasState(TriggerContext c) throws Exception {
```

Comment:
I believe this should return `ReadableState<Boolean>` and have a `prefetchHasState` so the caller can decide when to force the batched read.